lua/packer/luarocks.lua

```diff
@@ -40,11 +36,34 @@ else
   }
 end
 
-local cache_path = vim.fn.stdpath 'cache'
-local rocks_path = util.join_paths(cache_path, 'packer_hererocks')
-local hererocks_file = util.join_paths(rocks_path, 'hererocks.py')
-local hererocks_install_dir = util.join_paths(rocks_path, lua_version.dir)
-local shell_hererocks_dir = vim.fn.shellescape(hererocks_install_dir)
+local config = nil
+local cache_path = nil
```

Comment:
Slight nitpick: if we're making these externally configurable, let's group them together in a table to make them more manageable within this file. Really, I suppose they should just be fields in the `config` table, and we should use `vim.tbl_extend` with the appropriate merge behavior.

Reply:
Well, I try to invent something here, but it wouldn't be so easy: most parameters requires values of previously-defined value.

Current behaviour is just to check if it is defined in `config`, and use defined value if it is, and some default value if don't. And, as it is executed line-by-line, if some "early" parameter was defined in config, it will affect all next-defined things.

And if we'll just use a table, this magic won't work, as we cant call neither other fields in the table of "defaults", nor values from not-yet-merged resulting table.

So, I'll still need to check values from _config, or invent some other way.

// by the way, I found that `cache_path` here is never used in anu way except being a part of `rocks_path`, so I removed it as useless.

Hope, you'll not object that change.